internal/gensupport/send.go

```diff
@@ -98,10 +98,11 @@ func sendAndRetry(ctx context.Context, client *http.Client, req *http.Request, r
 		case <-ctx.Done():
 			// If we got an error, and the context has been canceled,
 			// the context's error is probably more useful.
```

Comment:
Would another possible solution to #1685 be simply to remove this entire block, and let the context raise the deadline exceeded error on its own?

Reply:
do you mean just `return nil, ctx.Err()`? otherwise, where would you suggest to put something equivalent to `case <-time.After(pause):` (the backoff)?

Reply:
@tritone Can you comment on this?

Reply:
Hey @quartzmo , thanks for flagging. I think the right answer is to return a wrapped error that includes both the context cancelation and the final error from the service, if available -- otherwise it can be pretty hard to debug either way. We implemented this in google-cloud-go: https://github.com/googleapis/google-cloud-go/blob/0da9ab0831540569dc04c0a23437b084b1564e15/internal/retry.go#L36

I can make a PR to implement this if folks agree.

Reply:
(Also I see that this PR already basically takes that approach -- happy to work with you on completing it yourself if you want; the main change I would ask for is to implement a custom error type that implements `Is(error) bool` so that we can preserve the validity of calls to `errors.Is` in user code.

Reply:
yep I'd like to fix that as you suggested, I will do it tomorrow

Reply:
sounds good, thank you!

Reply:
it's there, PTAL. I basically copied the `wrappedCallErr` struct from your other repo with the exception that I did `Is()` like this:

```
func (e wrappedCallErr) Is(target error) bool {
	return errors.Is(e.ctxErr, target) || errors.Is(e.wrappedErr, target)
}
```

basically leveraging `errors.Is()` instead of using plain `==` comparison. if that's too much pls let me know!

I also seized the opportunity to use this wrapper in all the other occurrences of some ctx+svc error